tests/rptest/tests/archive_retention_test.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2020 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from time import sleep
+import time
+from ducktape.errors import TimeoutError
+from ducktape.mark import matrix, parametrize
+from ducktape.utils.util import wait_until
+
+from rptest.services.kgo_verifier_services import KgoVerifierProducer
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import CloudStorageType, SISettings, MetricsEndpoint, get_cloud_storage_type
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.util import (produce_until_segments, produce_total_bytes,
+                         wait_for_local_storage_truncate, segments_count,
+                         expect_exception)
+from rptest.utils.si_utils import BucketView
+
+
+class CloudArchiveRetentionTest(RedpandaTest):
+    """Test retention with spillover enabled"""
+    def __init__(self, test_context):
+        self.extra_rp_conf = dict(
+            log_compaction_interval_ms=1000,
+            cloud_storage_spillover_manifest_max_segments=5)
+
+        si_settings = SISettings(test_context,
+                                 cloud_storage_housekeeping_interval_ms=1000,
+                                 log_segment_size=4096,
```

Comment:
nit: segment size will be rounded up to 1MiB unless you set `log_segment_size_min` too.